src/scripts/crank.ts

```diff
@@ -83,19 +84,39 @@ setInterval(
 
 
 async function run() {
-  const spotMarkets = await Promise.all(
-    markets[cluster].map((m) => {
-      return Market.load(
-        connection,
-        new PublicKey(m.address),
-        {
-          skipPreflight: true,
-          commitment: 'processed' as Commitment,
-        },
-        serumProgramId,
-      );
-    }),
-  );
+  let spotMarkets;
+  if (TOP_MARKET === 'false' || TOP_MARKET === undefined) {
+    spotMarkets = await Promise.all(
+      markets[cluster].map((m) => {
+        return Market.load(
+          connection,
+          new PublicKey(m.address),
+          {
+            skipPreflight: true,
+            commitment: 'processed' as Commitment,
+          },
+          serumProgramId,
+        );
+      }),
+    );
+  } else {
+    const { data } = await axios.get(
```

Comment:
can we please try-catch and retry? and also validate the results by checking if the the pubkeys are valid. We want the script to not break easily.

Reply:
adding try-catch.
how do we check if the public key is valid or not ??

Reply:
`new PublicKey("asdasd"); // fails` so maybe you're already validating

Reply:
yea, will throw error when its not valid